addons/medical_engine/functions/fnc_getItemArmour.sqf

```diff
@@ -0,0 +1,58 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Scans config or uses cached values to determine what armour a particular item provides to a particular hitpoint
+ *
+ * Arguments:
+ * 0: Item Classname <STRING>
+ * 1: Hitpoint <STRING>
+ * 2: Disable Caching <BOOLEAN> (Default: false)
+ *
+ * Return Value:
+ * Protection the item applies to that hitpoint <ARRAY>
+ * 0: Armour <NUMBER>
+ * 1: ExplosionShielding <NUMBER>
+ *
+ * Example:
+ * ["V_PlateCarrier_rgr", "HitChest"] call ace_medical_engine_fnc_getItemArmour
+ *
+ * Public: No
+ */
+
+ params ["_item","_hitpoint",["_noCache",false]];
+
+_hitpoint = toLower _hitpoint;
+
+// Return the cached value if found
+private _cached = if (_noCache) then {[]} else {GVAR(armourCache_items) getVariable [_item,[]]};
+if !(_cached isEqualTo []) exitwith {
+    private _index = _cached findIf {_x#0 isEqualTo _hitpoint};
+    if (_index == -1) then {[0,0]} else {(_cached select _index)#1};
+};
```

Comment:
The cache could be more powerful by merging the item classname and hitpoint as key using `format`. Then you don't need a loop.
```
getVariable [format ["%1$%2", _item, _hitpoint], []];
```
$ is not a valid char in classnames, thus safe to use a separator.

Reply:
I don't think this will work, for the same reason as below. E.g. when I look up "V_PlateCarrierGL_rgr$HitLegs" and find no value; does that mean I haven't cached it yet and need to scan the config, or I have checked it but it has no value for that hitpoint?